fdbserver/DDTeamCollection.actor.cpp

```diff
@@ -3014,21 +2999,22 @@ class DDTeamCollectionImpl {
 			// SOMEDAY: support wiggle multiple SS at once
 			ASSERT(!self->wigglingId.present()); // only single process wiggle is allowed
 
-			Optional<Value> localityKey;
-			Optional<Value> localityValue;
+			std::vector<std::pair<Optional<Value>, Optional<Value>>> localityKeyValues;
 			if (self->configuration.perpetualStorageWiggleLocality != "0") {
-				ParsePerpetualStorageWiggleLocality(
-				    self->configuration.perpetualStorageWiggleLocality, &localityKey, &localityValue);
+				localityKeyValues =
+				    ParsePerpetualStorageWiggleLocality(self->configuration.perpetualStorageWiggleLocality);
 			}
 
 			// if perpetual_storage_wiggle_locality has value and not 0(disabled).
-			if (localityKey.present()) {
-				if (self->server_info.count(res.begin()->first)) {
-					auto server = self->server_info.at(res.begin()->first);
-
-					// Update the wigglingId only if it matches the locality.
-					if (server->getLastKnownInterface().locality.get(localityKey.get()) == localityValue) {
-						self->wigglingId = res.begin()->first;
+			if (!localityKeyValues.empty()) {
+				for (const auto& [localityKey, localityValue] : localityKeyValues) {
```

Comment:
Can we move the for loop after the "auto server = self->server_info.at(res.begin()->first);"

Reply:
Good catch!

Reply:
Something like below would be better may be:

if (self->server_info.count(res.begin()->first)) {
     auto server = self->server_info.at(res.begin()->first);
     for (const auto& [localityKey, localityValue] : localityKeyValues) {
         / /Update the wigglingId only if it matches the locality.
         if (server->getLastKnownInterface().locality.get(localityKey.get()) == localityValue) {
 		self->wigglingId = res.begin()->first;

Reply:
Done